command/harness/dlite/handler.go

```diff
@@ -0,0 +1,43 @@
+package dlite
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/drone-runners/drone-runner-aws/command/harness"
+	"github.com/go-chi/chi"
+	"github.com/wings-software/dlite/client"
+	"github.com/wings-software/dlite/poller"
+)
+
+func Handler(p *poller.Poller) http.Handler {
```

Comment:
Can we move the http server code to dlite? Dlite can add a module that you can use to start the server. This handler will be present in dlite as well. Server code starter can take an interface 

type handler interface {
   EnableMaintenanceMode
   DisableMaintenanceMode
}

Everyone using dlite can implement this interface and start the server and use it.

Reply:
I thought about this initially but I would prefer to keep this here until we have a more generic abstraction for what maintenance mode means in dlite. I'm not sure yet how well this would translate to other runners. For now I have kept the filter logic in dlite which anyone can use and will move the maintenance mode logic to dlite as well once it's more generic.